adam-core/src/main/scala/org/bdgenomics/adam/hbase/HBaseFunctions.scala

```diff
@@ -0,0 +1,587 @@
+      hbaseTableName + "_meta",
+      vcRdd.sequences,
+      sequenceDictionaryId)
+
+    val rdd = if (partitions.isDefined) vcRdd.rdd.repartition(partitions.get)
+    else vcRdd.rdd
+
+    val genodata: RDD[(Array[Byte], List[(String, Array[Byte])])] = rdd.mapPartitions((iterator) => {
+      val genotypebaos: java.io.ByteArrayOutputStream = new ByteArrayOutputStream()
+      val genotypeEncoder = EncoderFactory.get().binaryEncoder(genotypebaos, null)
+
+      val genotypeDatumWriter: DatumWriter[Genotype] = new SpecificDatumWriter[Genotype](
+        scala.reflect.classTag[Genotype]
+          .runtimeClass
+          .asInstanceOf[Class[Genotype]])
+
+      val genotypesForHbase: Iterator[(Array[Byte], List[(String, Array[Byte])])] = iterator.map((putRecord) => {
+
+        val myRowKey = KeyStrategy1.getKey(KeyStrategy1rowKeyInfo(putRecord.variant.variant.getContigName,
+          putRecord.variant.variant.getStart.toInt,
+          putRecord.variant.variant.getEnd.toInt,
+          putRecord.variant.variant.getReferenceAllele,
+          putRecord.variant.variant.getAlternateAllele))
+
+        val genotypes: List[(String, Array[Byte])] = putRecord.genotypes.map((geno) => {
+          genotypebaos.reset()
+          genotypeDatumWriter.write(geno, genotypeEncoder)
+          genotypeEncoder.flush()
+          genotypebaos.flush()
+          (geno.getSampleId, genotypebaos.toByteArray)
+        }).toList
+
+        (myRowKey, genotypes)
+      })
+      genotypesForHbase
+    })
+
+    val bulkLoadFlatMap = (t: (Array[Byte], List[(String, Array[Byte])])) => {
+      val data = new ListBuffer[(KeyFamilyQualifier, Array[Byte])]
+      val rowKey = t._1
+      val family = Bytes.toBytes("g")
+      t._2.foreach((x) => {
+        val qualifier: String = x._1
+        val value: Array[Byte] = x._2
+        val mykeyFamilyQualifier = new KeyFamilyQualifier(rowKey, family, Bytes.toBytes(qualifier))
+        data += Tuple2(mykeyFamilyQualifier, value)
+      })
+
+      data.iterator
+    }
+
+    dao.hbaseBulkLoad(genodata,
+      hbaseTableName,
+      bulkLoadFlatMap,
+      stagingFolder)
+  }
+
+  /**
+   *
+   * @param dao HBase Data Access Object
+   * @param hbaseTableName HBase table name
+   * @param sampleIds List of sample ids
+   * @param sampleListFile File containing one sample id per row
+   * @param partitions number of paritions to use when performing bulk delete
+   */
+
+  def deleteGenotypeSamplesFromHBase(dao: HBaseSparkDAO,
+                                     hbaseTableName: String,
+                                     sampleIds: Option[List[String]] = None,
+                                     sampleListFile: Option[String] = None,
+                                     partitions: Option[Int] = Some(1)): Unit = {
+
+    val scan = new Scan()
+    scan.setCaching(100)
+    scan.setMaxVersions(1)
+
+    val sampleIdsFinal: List[String] =
+      if (sampleListFile.isDefined) {
```

Comment:
This should be rewritten as either a `getOrElse` or `fold` or some similar pattern.